docs/tutorials/craft-a-snap.rst

```diff
@@ -0,0 +1,605 @@
+snap needs less confinement to function, where it has fewer guardrails and greater
+access to the system.
+
+We want our snap to be as safe and secure as possible, so let's change these values to:
+
+.. literalinclude:: code/craft-a-snap/snapcraft.yaml
+    :language: yaml
+    :caption: snapcraft.yaml
+    :start-at: grade: stable
+    :end-at: confinement: strict
+
+Now, when we build the snap, the snap's access to the host is inverted -- all sensitive
+system resources are blocked unless facilitated by an interface. And when we publish the
+snap, users will be able to install it without the ``--devmode`` flag.
+
+
+Test the Interfaces
+-------------------
+
+Now that the snap is confined, we can test the interfaces realistically.
+
+Build and reinstall the snap, but this time, install it like a production-ready snap:
+
+.. literalinclude:: code/craft-a-snap/task.yaml
+    :language: bash
+    :start-at: snap install ukuzama-pyfiglet_0.1_amd64.snap --dangerous
+    :end-at: snap install ukuzama-pyfiglet_0.1_amd64.snap --dangerous
+    :dedent: 2
+
+.. note::
+
+    We must continue passing the ``--dangerous`` argument during installation because
+    it's not live in the Snap Store, and therefore not attestable.
+
+Next, let's gather a font that wasn't included with pyfiglet and install it.
+
+Download `Small Braille
+<https://github.com/xero/figlet-fonts/blob/master/smbraille.tlf>`_ from the figlet-fonts
+project and install it with:
+
+.. literalinclude:: code/craft-a-snap/task.yaml
+    :language: bash
+    :start-at: ukuzama-pyfiglet -L smbraille.tlf
+    :end-at: ukuzama-pyfiglet -L smbraille.tlf
+    :dedent: 2
+
+And give it a try:
+
+.. terminal::
+    :input: ukuzama-pyfiglet -f smbraille hamba kahle
+    :user: crafter
+    :host: home
+
+    ⣇⡀ ⢀⣀ ⣀⣀  ⣇⡀ ⢀⣀   ⡇⡠ ⢀⣀ ⣇⡀ ⡇ ⢀⡀
+    ⠇⠸ ⠣⠼ ⠇⠇⠇ ⠧⠜ ⠣⠼   ⠏⠢ ⠣⠼ ⠇⠸ ⠣ ⠣⠭
+
+
+Review the project file
+-----------------------
+
+Here's the complete code for the ukuzama-pyfiglet project. Yours should look similar to
+it.
+
+.. collapse:: snapcraft.yaml of ukuzama-pyfiglet
+
+    .. literalinclude:: code/craft-a-snap/snapcraft.yaml
+        :language: yaml
+
+
+Conclusion and next steps
+-------------------------
+
+And you're done! You have a snap of pyfiglet that works on your system.
+
+It would be a good time to start planning for your first public snap. Ask yourself, what
+software would be interesting to package? What apps would benefit the most from the
+security and ease of a snap? Any reason or justification is valid. Snaps can be tools,
+productivity software, games, or any traditional Linux package.
+
+Take a look on the public `Snap Store <https://snapcraft.io>`_ to see if the apps you
+use the most have public snaps. If a result comes up empty, that would be good candidate
+for your first public snap.
+
+When you're ready to begin crafting in earnest, you should :ref:`create an account and then register your snap <how-to-register-a-snap>`.
```

Comment:
@sergiusens We're wondering which path we want to send users down. Should we encourage them to snap their own software, or existing software?